### ISSUE TITLE
Enhancement: indicate when the original photo is viewed

### ISSUE THREAD

**Issue by elpraga** [owncloud/android#1243]
current beta

WIth the nice large thumnails that can be viewed in fullscreen, it would be nice to add a visual indication when we are viewing the actual file and not just a thumbnail to distinguish a bad quality original from the thumb especially when pinch zooming the preview.

**Comment by tobiasKaminsky** [comment]
I wanted to add a floating action button that allows one to quickly download the original file.
This could also serve as an indicator.
@jancborchardt What do you think?

**Comment by jancborchardt** [comment]
I’m not sure what purpose this would serve. The quality of image is for performance purposes only, so nothing relevant for the end user. I’d rather say that for example when needed (like on zooming in), the full quality image should automatically be lazy loaded.

**Comment by elpraga** [comment]
I as a user would still like to have some way of knowing if I am looking at the real picture of just a lower quality preview. And I would prefer to be in charge of downloading or not the original - it might make a difference on mobile data and in when using roaming, downloading large amount of date in the background might me an unpleasant surprise..

**Comment by jancborchardt** [comment]
> I as a user would still like to have some way of knowing if I am looking at the real picture of just a lower quality preview.

This should never really matter. People should always get served the best quality image for the purpose.

When the original image is then downloaded (on zooming or whatever) we could have an overlay below the header bar saying something like »Downloading high-res original …« with a cancel X on the right.

**Comment by elpraga** [comment]
> This should never really matter. People should always get served the best quality image for the purpose.

> When the original image is then downloaded (on zooming or whatever) we could have an overlay below the header bar saying something like »Downloading high-res original …« with a cancel X on the right.

This is nice (no mocking), but it does have two downsides that I think should be considered:
1. Mobile data usage, and especially roaming data. I really think, the user should make a conscious decision before starting to download 5MB picture, not just trying a pinch zoom.
2. Free space on a mobile device. Many times I am in a situation when I have to select photos and videos to erase to make free room, and it is a lengthy process. I can control the space used by the thumbs by setting the cache size, but if full resolution photos got downloaded automatically, I would have to to select again what to keep and what to detele. I also cannot delete everything and then download it again when I need it, because sometimes I am in places without coverage and I want to have certain files at hand.

Please, don't get me wrong, I understand that it would be very nice to do what You suggest, I just prefer having a choice. (At least until I can afford to buy a new phone and a new data plan any time I please :) )

**Comment by jancborchardt** [comment]
For point 1, that’s exactly what the notice up top is for:
> »Downloading high-res original …« with a cancel X on the right.

It doesn’t interfere and is easily cancelable. Also, it only happens when e.g. zooming, which basically is the inherent action for requesting a higher quality image. No need to have a separate button for that.

Good point with the free space though. But I guess this fits into the cache size as well? Or rather, the limit is your SD card and well, you have the actual file.

**Comment by elpraga** [comment]
> »Downloading high-res original …« with a cancel X on the right.

Last summer some roamings were counting by 1MB and pretty costly. Until I hit the cancel button I can have a whole 1MB of extra traffic. 

But I get what You mean, and it is really nice, what about a cuestion "Do You want to download the whole file?" when starting the pinch gesture?

> Or rather, the limit is your SD card and well, you have the actual file.
I am pointing it out, because I did play with the full screen thumbnails, and I really like it, and I did pinch and zoom, but I certainly did not want to download all the files I did the pinch and zoom in..

And one more thing to take in consideration is the time it takes to download the picture. It can easily be 10 seconds, if people are casually looking and zooming in and they don't know they started a download, they may easily go to another picture before the download is finished. And create a long download queue. You might pretty well end up with 15 downloads of full-size pictures without the user seeing any benefit of it if they don't know that pinching in starts the download. Just remember showing someone (or letting someone see) Your photos from the vacations. 

As I say, I do like the way You are thinking

**Comment by jancborchardt** [comment]
I can see the issue, but it _is_ an edge-case and at the same time something like a modal every time you zoom in to an image will disrupt the experience for _everyone_. So that’s not a feasible solution. I would suggest to not zoom in then when you are on roaming, or disable mobile networks/roaming as it’s pretty easy on Android.

After all, viewing and downloading images is what this app is for. And when you move to a different image you didn’t download yet, the behavior is exactly the same – it doesn’t ask you every time »Are you 100% sure you want to download this?« ;)

I do understand your issue. However keep in mind that there are thousands of people using this app, and we need to make sure that the main use cases and flows are best covered, and a solution to an edge-case does not interfere with it.

> if people are casually looking and zooming in and they don't know they started a download, they may easily go to another picture before the download is finished

Well – that’s why I’m saying there should be an overlay directly below the header. Maybe I wasn’t clear enough: This is not a notification you can easily miss in the notifications bar, it is a very visible toast-like overlay which will say »Downloading high-res original …«. If that is missed, there’s not much else we can do. ;)

**Comment by elpraga** [comment]
> »Downloading high-res original …«. If that is missed, there’s not much else we can do. ;)

:D :D :D OK. That should be fine, my concern was about large downloads without a clear notice.

**Comment by tobiasKaminsky** [comment]
I have a few remarks on this. 
The resized image that is stored into the cache is fullHD, which means it is a pretty good quality. 
Please open such an image and zoom in. If you do not do a 400% zoom, the quality will be still very good. --> So for zooming in a little it is not necessary to download the whole image.

For me this would be too much automation. If the user wants to have an original image, he should download it. (Same will be soon for videos/audio: Default is streaming, download requires an intentional action of the user).

I think the best would be to see the user acceptance of this new way, as there are some issues that demand "streaming" of photos, which is exactly this. 

(side note for the motivation of creating this enhancement: I have a phone with little disk, so I do not want to have any image downloaded in original size. Even if I share an image with whatsapp or email I do not want to sent it with original size as it would cost to much traffic).

Ps: If still, after some time, there is a need of automatically download the images when zooming it, it should at least be disabled completly: On first download when zooming show the toast with "downloading high-res image" and "stop" and "stop and never do it again!".

**Comment by jancborchardt** [comment]
Ok, or we do it like some maps apps: When you are zoomed in very far to the point where it would make sense to get a high-res version, show a small overlay with »Download high-res original?«

**Comment by jancborchardt** [comment]
Cause the alternative would be to put it in the dot-dot-dot menu in the right, but when you only rarely need the original it doesn’t really justify positioning it there.

**Comment by tobiasKaminsky** [comment]
My first idea was to use a floating action button to download the full sized image.
But I guess this is too much disturbing, especially if it is rarely used.

I like your idea with the zooming step, although I do not know if it is possible.
But we also need a download button, as (even if it is rarely) sometimes one want to have the full sized.
You say "it doesn’t really justify positioning it there". What would be your favourite position?

**Comment by jancborchardt** [comment]
Well, if the zooming step is difficult to do or not possible, the first step would indeed be to put it into the dot-dot-dot menu. The main problem I have with that is that every action there needs to be really thought through since otherwise this just ends up being the kitchen sink where everything is.

Kind of like the current long-press menu which is just unbelievably long to the point where it’s difficult to find the most important actions like share, download and favorite. For more of that see the issue:  »Actions in long-press menu« #1144

**Comment by elpraga** [comment]
Too bad that the zoom in a little and then offer a download is difficult to implement. It would be practical.

Even though I mostly agree with Tobias, that in most cases the previews are of sufficient quality, there are cases (like stitched panoramas) when I would opt for downloading the original file.

Could the download button appear on long tap or double tap?  Would that be better?

**Comment by jancborchardt** [comment]
Double-tap should be used for zooming. But long-tap on the image for downloading the full-res image seems like a good suggestion! What do you think @tobiasKaminsky @AndyScherzinger?

**Comment by tobiasKaminsky** [comment]
How would you inform the user that this is possible?
I think the most intuitive way is the "three dot" menu with "download full sized image".

**Comment by jancborchardt** [comment]
Well, not every functionality has to be directly known, right? For example, right now it is not possible at all to download the original, is it? As said, putting every single thing in the already crowded 3-dot menu will drown the really important actions.

**Comment by tobiasKaminsky** [comment]
Of course it is ;) Just click on "download" in the three dot menu.
As this menu is not visible at first, I do not think that it is a problem to put "all" entries that are useful in it...

**Comment by elpraga** [comment]
Ok, Today as 2015-11-05 we have menu and sharing with users!!! Waaayyyy!!! 
But when I click on a picture with downloaded original am I seeing the original or the large preview? We still don't have any indication of that and even when I am viewing the photo which has the original downloded I can zoom in muuuch less then for example using QuickPic on the same file..

**Comment by tobiasKaminsky** [comment]
For the zooming problem please open a new issue.
Do we really need an indicator?
Before you click on an image you see if it is downloaded. Is this not sufficient?
(Ok maybe when you swipe through an folder, you do not know it.)

But what is the purpose of knowing it? You can send the resized images just like a big one...
I do not think that it is necessary and it will clutter the UI (which @jancborchardt do not want and I totally agree with him).

**Comment by elpraga** [comment]
> (Ok maybe when you swipe through an folder, you do not know it.)

Yes, exactly, it is the case. And it is connected to viewing the fullscreen previews and zooming in. 

But my apologies. I double checked the behaviour, and the app does show the downloaded picture in fullscrenn when available and the menu entry "Download" changes to "Open with".

I think this is sufficient indication if I am viewing the original or not.

**Comment by jancborchardt** [comment]
Cool! :)
